Modules/@babylonjs/react-native/BabylonModule.ts

```diff
@@ -1,33 +1,72 @@
 import { NativeModules } from 'react-native';
-import { NativeEngine } from '@babylonjs/core';
+import { Engine, NativeEngine } from '@babylonjs/core';
 
-// This global object is part of Babylon Native.
+const disposedPropertyName = "BabylonReactNative_IsDisposed";
+
+declare const global: {
+    nativeCallSyncHook: any;
+};
+const isRemoteDebuggingEnabled = !global.nativeCallSyncHook;
+
+// This global object is owned by Babylon Native.
 declare const _native: {
     whenGraphicsReady: () => Promise<void>;
-    engineInstance: NativeEngine;
-}
+};
+
+// This JSI-based global object is owned by Babylon React Native.
+// This will likely be converted to a TurboModule when they are fully supported.
+declare const BabylonNative: {
```

Comment:
>BabylonNative [](start = 14, length = 13)

 is BabylonNative supposed to be a native module or a jsi/napi constructed object? Is it fair to assume that all constants loaded via NativeModules.* are react-native native modules while constants not loaded via NativeModules.* are jsi/napi populated?

Reply:
By `NativeModules.*`, do you mean on the JS side, such as:

```javascript
const BabylonModule: {
    initialize(): Promise<void>;
} = NativeModules.BabylonModule;
```

If that is what you mean by `NativeModules.*`, then those are instantiated by React Native itself. `BabylonNative` is a JSI `HostObject` that we manually construct and inject into the JS execution context (which I think is what you mean by jsi/napi constructed/populated). If that is all correct, then the answer to your question is yes.

In the future, we should be able to get rid of `BabylonModule`. When TurboModules are fully supported, a TurboModule is itself just a derivation of `HostObject`, so transitioning the `BabylonNative` object to a TurboModule (when fully supported) should be straight forward (and will let us remove `BabylonModule`).

Reply:
Chatted offline, this makes a lot more sense now. It sounds like NativeModules.* and requireNativeComponent is following what's expected for react-native